### PR TITLE
[Android] Use NDK toolchain instead of CMake for library mode

### PR DESCRIPTION
**Pull request by steveisok**
This change switches from relying on CMake to clang directly for Android library mode. The intent is to keep the number of build dependencies to a minimum.

### CODE REVIEW

File: src/tasks/Common/Utils.cs [hunk 1]
```diff
@@ -354,6 +354,27 @@ public static bool IsWindows()
 #endif
     }
 
+<<<<<<< HEAD
```
Comment (radical) [inline src/tasks/Common/Utils.cs:357]: Merge markers

File: src/mono/msbuild/android/build/AndroidBuild.targets [hunk 2]
```diff
@@ -47,6 +46,9 @@
     </PropertyGroup>
 
     <ItemGroup Condition="'$(_IsLibraryMode)' == 'true'">
+      <_CommonLinkerArgs Include="-l:libz.so;-l:liblog.so;-l:libc.so;-l:libm.so;" />
```
Comment (akoeplinger) [inline src/mono/msbuild/android/build/AndroidBuild.targets:49]: nit: I think it'd be more readable like this:
```suggestion
      <_CommonLinkerArgs Include="-l:libz.so" />
      <_CommonLinkerArgs Include="-l:liblog.so" />
      <_CommonLinkerArgs Include="-l:libc.so" />
      <_CommonLinkerArgs Include="-l:libm.so" />
```

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**